library/src/doo/shell.clj

```diff
@@ -37,13 +37,20 @@
       (.close r)
       (.toString out))))
 
-(defn exec! [cmd]
+(defn -exec
+  ([command-arr]
+    (.exec (Runtime/getRuntime) command-arr))
+  ([command-arr working-dir]
+   (.exec (Runtime/getRuntime) command-arr nil working-dir)))
+
+(defn exec! [cmd exec-dir]
   (let [windows? (= :windows (utils/get-os))
-        windows-cmd (when windows? ["cmd" "/c"])]
+        windows-cmd (when windows? ["cmd" "/c"])
+        -exec (if exec-dir #(-exec % exec-dir) -exec)]
```

Comment:
I don't really like shadowing the existing function names, perhaps exec-fn would be more descriptive here?